source/blog/2016-09-21-running-tempest-on-rdo-openstack-newton.html.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Running Tempest on RDO OpenStack Newton
+author: chandankumar
+tags: rdo,packstack,tempest,newton,packaging
+date: 2016-09-21 11:23:29 IST
+---
+
+
+[Tempest](http://docs.openstack.org/developer/tempest/overview.html) is a set of integration tests to run against an OpenStack cluster.
+
+**What does [RDO](https://www.rdoproject.org) provides for Tempest?**
+
+RDO provides three packages for running tempest against any OpenStack installation.
+
+* *python-tempest* : It can be used as a python library, consumed as a dependency for out of tree tempest plugins i.e. for horizon and designate tempest plugins.
+* *openstack-tempest* : It provides python tempest library and required executables for running tempest.
+* *openstack-tempest-all* : It will install openstack-tempest as well as all the tempest plugins on the system.
+
+**Deploy [packstack](github.com/openstack/packstack) using latest RDO Newton packages**
+
+Roll out a vm of [CentOS 7](https://www.centos.org/download/), Follow these steps:
+
+1. Install rdo-release-newton rpm
+
+        # yum -y install https://rdoproject.org/repos/openstack-newton/rdo-release-newton.rpm
+
+2. Update your centos vm and perform reboot.
+
+	    # yum -y update
+
+3. Install openstack-packstack
+
+	    # yum install -y openstack-packstack
+
+4. Run packstack by enabling RDO GA testing repo:
+
+	    # packstack --enable-rdo-testing=y --allinone
+
+   Once packstack installation is done, we are good to go ahead.
+
+**Install tempest and required tempest plugins**
+
+ 1. Install tempest
+
+	    # yum install openstack-tempest
+
+2. Install tempest plugins based on the openstack services installed and configured on deployment.
+
+   Packstack installs by default horizon, nova, neutron, keystone, cinder, swift, glance, ceilometer, aodh and gnocchi.
+   To find out what are the openstack components installed, just do a rpm query:
+
+	    # rpm -qa | grep openstack-*
+
+	OR you can use openstack-status command for the same.
+	Just grab the tempest plugins of these services and install it.
+
+	    # yum install python-glance-tests python-keystone-tests python-horizon-tests-tempest \
+	      python-neutron-tests python-cinder-tests python-nova-tests python-swift-tests \
+	      python-ceilometer-tests python-gnocchi-tests python-aodh-tests
+
+	OR you can simply download install_test_packages.py script to install tempest plugins and run it.
+	It will automatically installs the required tempest plugin test packages.
+
+	    # python /usr/share/openstack-tempest-*/tools/install_test_packages.py
+
+
+3. To find what are tempest plugins installed:
+
+	    # tempest list-plugins
+
```

Comment:
Yep, exactly! Running 'epi group show tempest.cm' would show you the entry points for the plugins too.